server/app/lib/utils/fusor/config_template_utils.rb

```diff
@@ -0,0 +1,119 @@
+          create_rhevm_guest_agent_snippet
+        else
+          ::Fusor.log.error "====== Unsupported custom snippet '#{name}'.  Cannot ensure snippet. ======"
+          return nil
+        end
+      end
+
+      def ensure_kickstart(name)
+        if !ConfigTemplate.find_by_name(name).nil?
+          ::Fusor.log.debug "====== Template '#{name}' already exists! ======"
+          return
+        end
+
+        default_template_name = 'Satellite Kickstart Default'
+        default_template = ConfigTemplate.find_by_name(default_template_name)
+        if default_template.nil?
+          ::Fusor.log.error "====== Template '#{default_template_name}' does not exist! Cannot clone from default! ======"
+          return nil
+        end
+        new_template = default_template.clone
+        new_template.name = name
+        str = new_template.template.clone
+        str.sub! "name: #{default_template_name}", "name: #{name}"
+        new_template.template = str
+        new_template.save!
+      end
+
+      def append(name, text)
+        template = ConfigTemplate.find_by_name(name)
+        if template.nil? || text.nil?
+          ::Fusor.log.error "====== Input params is nil! Cannot append. ======"
+          return nil
+        end
+        str = template.template.clone
+        str.concat(text)
+        template.template = str
+        template.save!
+      end
+
+      def update(name, replacement_text, search_text)
+        template = ConfigTemplate.find_by_name(name)
+        if template.nil? || replacement_text.nil? || search_text.nil?
+          ::Fusor.log.error "====== Input params is nil! Cannot update ======"
+          return nil
+        end
+
+        if !template.template.include?(search_text)
+          ::Fusor.log.error "====== Search text '#{search_text}' does not exist in the template! Cannot update ======"
+          return nil
+        end
+        str = template.template.clone
+        str.sub! search_text, replacement_text
+        template.template = str
+        template.save!
+      end
+
+      def ensure_ks_with_snippet(ks_name, snippet_name)
+        return nil unless ensure_kickstart(ks_name)
+        return nil unless ensure_snippet(snippet_name)
+
+        # add snippets just before the 'sync' in the post section of ks
+        search_text = "\nsync\n"
+        snippet_include_line = "\n\n<%= snippet '#{snippet_name}' %>\n\n"
+
+        ks = ConfigTemplate.find_by_name(ks_name)
+        if ks.template.include?(snippet_include_line)
+          ::Fusor.log.debug "====== Snippet '#{snippet_name}' already in the KS '#{ks_name}'.  Nothing to do. ======"
+          return
+        end
+        replacement_text = snippet_include_line + search_text
+        update(ks_name, replacement_text, search_text)
+      end
+
+      #########################
+      # create custom snippets
+      #########################
+      def create_rhevm_guest_agent_snippet
+        name = @rhevm_guest_agent_snippet_name.clone
```

Comment:
Small comment of usage of .clone here.

I don't think this needs to be a .clone, I didn't see you making any changes to the 'name', just reading it and passing into append() to do a find/lookup.